java/src/jmri/jmrit/logix/NXFrame.java

```diff
@@ -224,51 +196,77 @@ private JPanel makeSwitchPanel() {
         return panel;
     }
 
-    private JButton getButton(String text) {
-        JButton button = new JButton();
-        button.setFont(new Font(Font.SANS_SERIF, Font.PLAIN, 12));
-        button.setText(text);
-        button.setSelected(true);
-        int bWidth = new JTextField(2).getPreferredSize().width;
-        int bHeight = new JTextField(2).getPreferredSize().height;
-        button.setMaximumSize(new Dimension(bWidth, bHeight));
-        return button;
-    }
-
     @Override
     protected void maxThrottleEventAction() {
         NumberFormat formatter = NumberFormat.getNumberInstance(); 
         float num = 0;
         try {
-            num =  formatter.parse(_maxThrottleBox.getText()).floatValue()/100;
-            num = Math.min(1.0f, Math.max(num,  0.f));
-            _maxThrottleBox.setText(formatter.format(num*100));
+            num =  formatter.parse(_maxThrottleBox.getText()).floatValue();
+            num = Math.min(100.0f, Math.max(num,  0.f));
+            _maxThrottleBox.setText(""+num);
         } catch (java.text.ParseException pe) {
             _maxThrottleBox.setText(null);
             _maxSpeedBox.setText(null);
             return;
         }
-        _speedUtil.setIsForward(_forward.isSelected());
-        float speed = _speedUtil.getTrackSpeed(num);
-        if (_speedUnits.getText().equals(LABEL_MPH)) {
-            _maxSpeedBox.setText(formatter.format(speed * _scale * 2.2369363f));                        
+        float speed = _speedUtil.getTrackSpeed(num/100);    // returns mm/ms
+        switch(_displayPref) {
+            case MPH:
+                _maxSpeedBox.setText(formatter.format(speed * _scale * 2.2369363f));
+                break;
+            case KPH:
+                _maxSpeedBox.setText(formatter.format(speed * _scale * 3.6f));
+                break;
+            case MMPS:
+                _maxSpeedBox.setText(formatter.format(speed * 1000));  // mm/sec
+                break;
+            case INPS:
+            default:
+                _maxSpeedBox.setText(formatter.format(speed * 39.37f));  // in/sec
```

Comment:
Please document these constants. What is the number 2.2369363? What is 3.6? What is 39.37? The next developer may not understand what they stand for. These numbers are used in several places in the file so I recommend creating a constant:

`public static final float SOMETHING = 123.456f;     // This constant is PI * 2`